app/src/main/java/org/oppia/app/utility/ClickableAreasImage.kt

```diff
@@ -21,29 +19,32 @@ class ClickableAreasImage(
   private val imageView: ImageRegionSelectionInteractionView,
   private val parentView: FrameLayout,
   private val listener: OnClickableAreaClickedListener
-) : OnPhotoTapListener {
-  private val attacher: PhotoViewAttacher = PhotoViewAttacher(imageView)
-
+) {
+  private var rect = Rect()
```

Comment:
Is this actually needed? It seems like the code can be simplified some:

```kotlin
  /** Get X co-ordinate scaled according to image.*/
  private fun getXCoordinate(contentWidth: Int, x: Float): Float {
    return x * contentWidth
  }

  /** Get Y co-ordinate scaled according to image.*/
  private fun getYCoordinate(contentHeight: Int, y: Float): Float {
    return y * contentHeight
  }
```

And then pass in the content width/height when computing the coordinates. Note that this seems to work since you're setting the left/top of the rect to 0, anyway. You also really only need the rect in the function it's being created, so we can omit it entirely to simplify the implementation.

Reply:
I simplified it a bit more 
`
  /** Get X co-ordinate scaled according to image.*/
  private fun getXCoordinate(x: Float): Float {
    return x * getImageViewContentWidth()
  }

  /** Get Y co-ordinate scaled according to image.*/
  private fun getYCoordinate(y: Float): Float {
    return y * getImageViewContentHeight()
  }

`

Reply:
You didn't actually remove ``rect``--please do so.